src/Markup/Avalonia.Markup.Xaml/MarkupExtensions/CompiledBindings/NotifyingPropertyInfoHelpers.cs

```diff
@@ -0,0 +1,255 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Text;
+using Avalonia.Data.Core;
+using Avalonia.Reactive;
+using Avalonia.Utilities;
+
+namespace Avalonia.Markup.Xaml.MarkupExtensions.CompiledBindings
+{
+    public static class NotifyingPropertyInfoHelpers
```

Comment:
I'm going to work on a slightly different design for this portion since I had a few ideas going when I first wrote it and ended up with a weird amalgamation of them.